openrtb_ext/request_wrapper.go

```diff
@@ -72,6 +72,18 @@ func (rw *RequestWrapper) LenImp() int {
 	return len(rw.Imp)
 }
 
+// SyncImps synchronizes impWrappers with reqWrapper.BidRequest.Imp using the last one as a source of truth
+func (rw *RequestWrapper) SyncImps() {
+	if len(rw.Imp) > 0 {
+		rw.impWrappers = make([]*ImpWrapper, len(rw.Imp))
+		for i := range rw.Imp {
+			rw.impWrappers[i] = &ImpWrapper{Imp: &rw.Imp[i]}
+		}
+	} else {
+		rw.impWrappers = nil
+	}
+}
+
```

Comment:
@hhhjort may you please confirm this approach looks right to you, and I'll add unit tests.
To fix the issue we need a direct and unconditional access to the private impWrapper. I was able to make the request working using GetImp() and SetImp(), but many existing unit tests failed because of 
```
expected: []Imp(nil)
actual: []Imp{}
```
And also this code looks cleaner and straight forward.

Reply:
I would suggest modifying `rebuildImp()` to set `Imp` to nil if the list of imps has a zero length. We only use the raw request as a source of truth on wrapper creation, I would rather not break that convention. Once we allow it to go the other way once, it opens the door for others to do it as well, and then things can start getting confusing as some code pushes changes from the request to the. wrapper, and others in the opposite direction. I would rather fix things such that the wrapper works consistently.

Reply:
I tried to add changes there first. `rebuildImp` takes `rw.impWrappers` as a source of truth and have conditions that returns.

```go
func (rw *RequestWrapper) rebuildImp() error {
	if !rw.impWrappersAccessed {  -> we always need to be sure this flag is true in order not to return
		return nil
	}

	if rw.impWrappers == nil { -> we need to have it nil in our case to set rw.Imp = nil, but it never is and there is no way to set it to nil
		rw.Imp = nil
		return nil
	}

        // we need to add logic of synchronization between reqWrapper.BidRequest.Imp and reqWrapper.impWrappers
        // but how to determine what is a source of truth?


	rw.Imp = make([]openrtb2.Imp, len(rw.impWrappers)) 
	for i := range rw.impWrappers {
		if err := rw.impWrappers[i].RebuildImp(); err != nil {
			return err
		}
		rw.Imp[i] = *rw.impWrappers[i].Imp
	}

	return nil
}
```